### PR TITLE
Fix kernel `do_install` not singing kernel image

### PR DESCRIPTION
**Pull request by fedefrancescon**
With a full rebuild from scratch I noticed kernel `do_install` failed signing the kernel image.
`mkimage_imx8.csf` doesn't exists, it should be `mkimage_imx8`. This seems to be broken since 24 Aug.
Probably we didn't noticed before because we were using "cached" tools build.

### CODE REVIEW

**Comment by fedepell** [comment]
I've checked this better now. The `mkimage_imx8.csf` tool is indeed installed by the `imx-boot` receipe. You can see it here:  https://github.com/fedepell/meta-engicam-nxp/blob/kirkstone-gwc/recipes-bsp/imx-mkimage/imx-boot_1.0.bbappend

This is done because `mkimage_imx8` (the "original" one) would be installed too late by default, at deploy stage. Here it is therefore installed before, so the kernel can use it (as the kernel as well needs it before) for signing.

So as far as `imx-boot` is built before it should all be fine as it is. It should be as well built before since the kernel depends on it (see [here]https://github.com/fedepell/meta-engicam-nxp/blob/68007a623e54bf44a4cdef4718ff823244125292/recipes-kernel/linux-fslc-imx/linux-fslc-imx_5.10.bbappend#L12).

So in short: to me it all seems fine and makes sense, modulo strange bugs, as far as `csfsigned` and its parameters are well defined. So I will not merge this but will leave it open to retest it from a total scratch build as soon as I have a bit more time or better when the necessity arises (ie. with kirkstone update?).

**Comment by fedefrancescon** [comment]
I've just tried to do a full rebuild from scratch and the issue is the very same.  
I think the issue is that the csf utilities are hardcoded and precompiled. If you recall there is a custom variable `+CSFPATH = "/srv/yocto-hsc/cst-3.1.0"` set in `conf/machine/imx8xq-smarcore.conf`

Here is a short log:

```
| -b /tmp/depmod.1lN2uU 5.10.109-HSC-1+
| ln: failed to create symbolic link '/lib/modules/99.98.5.10.109-HSC-1+': Permission denied
| -ae -F System.map 99.98.5.10.109-HSC-1+
| /srv/yocto-hsc/fsl-community-bsp/build/tmp/work/imx8xq_smarcore-fslc-linux/linux-fslc-imx/5.10.109-HSC-1-r1/temp/run.do_install.1505079: line 168: /srv/yocto-hsc/fsl-community-bsp/build/tmp/deploy/images/imx8xq-smarcore/imx-boot-tools/mkimage_imx8.csf: No such file or directory
| WARNING: /srv/yocto-hsc/fsl-community-bsp/build/tmp/work/imx8xq_smarcore-fslc-linux/linux-fslc-imx/5.10.109-HSC-1-r1/temp/run.do_install.1505079:168 exit 127 from '/srv/yocto-hsc/fsl-community-bsp/build/tmp/deploy/images/imx8xq-smarcore/imx-boot-tools/mkimage_imx8.csf -soc QX -rev B0 -c -ap arch/arm64/boot/Image a53 0x80280000 --data arch/arm64/boot/dts/engicam/imx8xq-icore-starterkit.dtb a53 0x83000000 -out /srv/yocto-hsc/cst-3.1.0/flash_os.bin'
| WARNING: Backtrace (BB generated script):
| 	#1: do_install, /srv/yocto-hsc/fsl-community-bsp/build/tmp/work/imx8xq_smarcore-fslc-linux/linux-fslc-imx/5.10.109-HSC-1-r1/temp/run.do_install.1505079, line 168
| 	#2: main, /srv/yocto-hsc/fsl-community-bsp/build/tmp/work/imx8xq_smarcore-fslc-linux/linux-fslc-imx/5.10.109-HSC-1-r1/temp/run.do_install.1505079, line 292
ERROR: Task (/srv/yocto-hsc/fsl-community-bsp/sources/meta-freescale/recipes-kernel/linux/linux-fslc-imx_5.10.bb:do_install) failed with exit code '1'
NOTE: Tasks Summary: Attempted 4334 tasks of which 4295 didn't need to be rerun and 1 failed.
NOTE: Writing buildhistory
NOTE: Writing buildhistory took: 15 seconds

Summary: 1 task failed:
  /srv/yocto-hsc/fsl-community-bsp/sources/meta-freescale/recipes-kernel/linux/linux-fslc-imx_5.10.bb:do_install
Summary: There was 1 WARNING message.
Summary: There was 1 ERROR message, returning a non-zero exit code.
```

There is also a minor issue: `ln: failed to create symbolic link '/lib/modules/99.98.5.10.109-HSC-1+': Permission denied` So probably a variable expansion error. This may be not a "real" issue. don't know.

**Comment by fedepell** [comment]
Can you manually just `bitbake imx-boot` first and see if then that file appears as explained above?

**Comment by fedefrancescon** [comment]
> bitbake imx-boot

This actually succeeds. After that I tried to redo `bitback engicam-gwc` and the issue is the same.

Is it possible that we are using the wrong path?

**Comment by fedefrancescon** [comment]
Found the real issue!

After your support I was able to identify that was a state cache/build cache issue! Some some (yet unknown) reasons, removing all build files was not enough. Seems that the custom installed files were not considered to be updated.

For further reference, I'm not 100% sure all this is really needed but, to properly do a (fully) clean build I used the following commands:

```bash
bitbake -fc cleanall engicam-gwc
rm -rf ~/fsl-community-bsp/build/{tmp,cache,sstate-cache}
```

Closing for now! Thanks
